packit/api.py

```diff
@@ -61,7 +61,7 @@ def sync_release(self, dist_git_branch: str, version: str = None):
 
         dg = DistGit(config=self.config, package_config=self.package_config)
 
-        full_version = version or up.get_upstream_version()
+        full_version = version or up.get_specfile_version()
```

Comment:
I am not sure if we really want this. Can this be configurable?

The version from the spec is useful when we do not want to edit specfile, but if there are packages that only needs to bumb version (and add changelog), then we can:
- get the version from rebase_helper
- save the version to specfile
- our reverse-sync will get the change backwards to the upstream repo

(Maintainer does not need to change the version in specfile by hand.)

Reply:
> I am not sure if we really want this. Can this be configurable?

I don't follow. Where do you want to get the version from then? Not every package is in URM. Spec file is the only trusted source for us (alternatively, get it from CLI, but that has no use for the packit service).

> The version from the spec is useful when we do not want to edit specfile, but if there are packages that only needs to bumb version (and add changelog), then we can:
> 
> * get the version from rebase_helper

Which means get it from specfile.

> * save the version to specfile
> * create a PR
> * our reverse-sync will get the change backwards to the upstream repo (after the merge)

I don't follow.

> (Maintainer does not need to change the version in specfile by hand.)

But they should, the version in spec should be up to date, otherwise packit won't work.

Feel free to PM me your thoughts in the language of our tribe.

Reply:
>  Where do you want to get the version from then? 

URM

> Not every package is in URM. Spec file is the only trusted source for us (alternatively, get it from CLI, but that has no use for the packit service).

I would try to use the following approach to get the version:
1. version from the parameter (from fedmsg or CLI option)
2. Try URM
3. Fallback to Specfile

We can also test the difference between URM and Specfile.

>> - get the version from rebase_helper
>
> Which means get it from specfile.

I meant "use URM through rebase-helper".

> But they should, the version in spec should be up to date, otherwise packit won't work.

I would like to automate that..;-)
If we have also the donwstream->upstream sync, we can change the version in downstream specfile and it will be propagated to upstream lately.

> I don't follow.
> :
> Feel free to PM me your thoughts in the language of our tribe. smiley

We can discuss that in person on Thursday.

Reply:
Just thought about this:

> We can also test the difference between URM and Specfile.

We can definitely check, but I wouldn't do anything about it: for people who are not using `git describe` style versioning, a common practice is to bump version after a release - different version in spec and URM - not an error.

I agree with your priority list though.